### PR TITLE
[Mobile Payments] Specific tap on mobile errors

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #8295 

## Description

In sTAP Away, we're adding support for Apple's built in card readers to take in-person payments. We continue to use Stripe's SDK to handle payments.

Stripe's connection flow is technically very similar to the bluetooth reader flow, but conceptually, there are differences and it's important that we make it clear to users that they are using the built in reader, and don't give the wrong impression that they are connecting to a bluetooth reader, to avoid confusion. Part of that is to display relevant errors and tips to resolve them.

This PR adds the specific error codes which can be produced by the Stripe SDK related to tap on mobile flows. These will show if card reader discovery fails for these reasons. They will show in a generic error screen: #8082 includes tasks to make specific screens for some of these errors.

## Testing instructions

🚨 I've been unable to test all of these errors, as iOS wouldn't let me remove the passcode from my device(!)

### Using an iPhone XS or newer on iOS 15.7 or older

1. Navigate to `Menu > Settings > Experimental features`
2. Turn on `Tap to Pay on iPhone`
3. Navigate to `Menu > Payments > Collect payment`
4. Go through the payment flow, and select `Card` on the payment method screen
5. When asked for a reader type, tap `Tap to Pay on iPhone` and go through the Terms of Service Apple ID linking (if you've not done so before)
6. Observe that you see an error message telling you that you need to use iOS 16 or newer

### Using an iPhone XS or newer on iOS 16.0 without a passcode set

Repeat the above steps, and observe that you see an error message advising you to set a passcode

### Using an iPhone XS or newer on iOS 16.0 with a passcode which is not signed in to iCloud

Repeat the above steps, and observe that you see an error message advising you that you need to sign in to iCloud.

Note that this error should only show for a merchant account which has not had Apple's terms of service accepted.

N.B. There are known issues with the payment flow after connection. In particular, #8289, #8288, #8274, and #8085. The payment flow "works", but these issues aren't addressed yet, and some are pre-existing in the legacy flow (just a lot less obvious when you use a bluetooth reader.)

Please do note any issues you find, but give me a shout if you want to check they're known first, just to save writing out detailed repro steps.

## Screenshots
unsupported-device

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalProgressDisplaying.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import UIKit
+
+protocol CardPresentModalProgressDisplaying: CardPresentPaymentsModalViewModel {
```
Comment (koke) [inline WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalProgressDisplaying.swift:3]: This part doesn't seem related to errors or mentioned in the description/testing, I'm wondering if this was intentionally included

**Review by koke — APPROVED** [review]
I was only able to test the iOS 15 case, since logging out of iCloud on my main device would be very inconvenient, but the code looks good.

File: WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalProgressDisplaying.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalProgressDisplaying.swift:3]: Github's being a bit strange.

Here's the first four commits from this PR:
CleanShot 2022-12-07 at 13 03 04@2x

And here's the three commits from the previous PR, which is merged to trunk:
CleanShot 2022-12-07 at 13 03 54@2x

So... they're all already in trunk with matching hashes, so I'm pretty sure this will sort itself out when I merge this PR... but I will check carefully! Thanks for the heads-up.
